lib/Importer/ONNXModelLoader.cpp

```diff
@@ -777,6 +777,55 @@ ONNXModelLoader::getTensorType(const ONNX_NAMESPACE::ValueInfoProto &in) {
   return Type(kind, dim);
 }
 
+Error ONNXModelLoader::getInputsNamesAndTypes(
+    std::vector<std::string> &inTensorNames, std::vector<Type> &inTypes,
+    const std::string &filename) {
+  // Creating GraphProto from modelfile
+  ONNX_NAMESPACE::ModelProto modelDef;
+
+  ASSIGN_VALUE_OR_RETURN_ERR(modelDef, loadProto(filename, false, nullptr));
+
+  ONNX_NAMESPACE::GraphProto graphDef = modelDef.graph();
+
+  // GraphDef.input can have both inputs and intermediate tensors whereas
+  // initializers have info about intermediate tensors. Thus the difference
+  // betweem two is taken
+  std::vector<std::string> inputs;
+  for (auto &in : graphDef.input()) {
+    inputs.push_back(in.name());
+  }
+
+  for (const auto &in : graphDef.initializer()) {
+    std::vector<std::string>::iterator position =
+        std::find(inputs.begin(), inputs.end(), in.name());
+    if (position != inputs.end()) {
+      inputs.erase(position);
+    }
+  }
+
+  for (const auto &finalIn : inputs) {
+    for (const auto &in : graphDef.input()) {
+      if (finalIn.compare(in.name()) == 0) {
```

Comment:
nit: early continue to remove a level of indentation:
```suggestion
      if (finalIn.compare(in.name()) != 0) {
        continue;
      }
```

Reply:
Done